LambdatestEcom/Pages/ProductPage.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DecoratorDesignPatternTests.Models;
+using LambdatestEcom.Pages;
+using Microsoft.Playwright;
+
+namespace LambdatestEcom.Pages
+{
+    internal class ProductPage
+    {
+        private readonly IPage _page;
+
+        public ProductPage(IPage page)
+        {
+            _page = page;
+        }
+
+        public async Task IncreaseQuantityOfProduct(int quantity)
+        {
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Increase quantity" }).ClickAsync();
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Increase quantity" }).ClickAsync();
+            //await _page.GetByRole(AriaRole.Button).Filter(new() { HasText = "Increase quantity" }).ClickAsync();
+            //await _page.GetByRole(AriaRole.Button).Filter(new() { HasText = "Increase quantity" }).ClickAsync();
+            
+        }
+         public async Task AddProductToCart()
+         {
+            //await _page.GetByRole(AriaRole.Button, new() { Name = "Add to Cart" }).ClickAsync();
+            await _page.Locator("#entry_216842").GetByRole(AriaRole.Button, new() { Name = "Add to Cart", Exact = true }).ClickAsync();
```

Comment:
Locator("#entry_216842") will work only with one product. AddProductToCart will fail for any other product

Reply:
Ok, thanks, fixed.